### PR TITLE
Use of triple point of water in GetSatVapPres 

### PR DESCRIPTION
**Pull request by dmey**
This PR introduces the use of triple point of water in `GetSatVapPres` to make the functions more physically correct and simplifies considerably the NR convergence procedure in `GetTDewPointFromVapPres`. This issue arises from the discontinuity at the freezing point of water.

### CODE REVIEW

**Comment by dmey** [comment]
@didierthevenard should the other couple of instances of freezing point of water also be replaced? 
E.g.

https://github.com/psychrometrics/psychrolib/blob/af24c1c3ed9079823faa7026d2d3e0cce571bd3c/src/python/psychrolib.py#L590

If not, to make the code cleared we should probably make this into a parameter and call it something like `FREEZING_POINT_WATER_{SI, IP}`

File: src/python/psychrolib.py [hunk 1]
```diff
@@ -949,6 +940,12 @@ def GetSatVapPres(TDryBulb: float) -> float:
 
     Reference:
         ASHRAE Handbook - Fundamentals (2017) ch. 1  eqn 5 & 6
+        Important note: the ASHRAE formulae are defined above and below the freezing point but have
```
Comment (didierthevenard) [inline src/python/psychrolib.py]: For completeness, this comment should be copied to all other languages.

**Comment by didierthevenard** [comment]
Regarding comment https://github.com/psychrometrics/psychrolib/pull/36#issuecomment-516375909: I have checked, and please tell me if I am wrong, but I can find only one place where the freezing point of water is used, that is, in GetHumRatioFromTWetBulb(). I have checked and the formulae in that function do present a discontinuity at the freezing point. However changing the threshold to the triple point does not resolve the discontinuity so I am not sure if this is related at all to the issue we had in GetSatVapPres() - likely not. So no change needed. The discontinuity does not seem to cause any issues anyway. As for changing the values 32 and 0 to FREEZING_POINT_WATER I think it would be a good idea. For consistency I suggest that the constant could be declared at the same time as TRIPLE_POINT_WATER, despite the fact that it is used only in one function and could be declared there locally instead.

**Comment by dmey** [comment]
>Regarding comment #36 (comment): I have checked, and please tell me if I am wrong, but I can find only one place where the freezing point of water is used, that is, in GetHumRatioFromTWetBulb(). I have checked and the formulae in that function do present a discontinuity at the freezing point. However changing the threshold to the triple point does not resolve the discontinuity so I am not sure if this is related at all to the issue we had in GetSatVapPres() - likely not. So no change needed. The discontinuity does not seem to cause any issues anyway. As for changing the values 32 and 0 to FREEZING_POINT_WATER I think it would be a good idea. For consistency I suggest that the constant could be declared at the same time as TRIPLE_POINT_WATER, despite the fact that it is used only in one function and could be declared there locally instead.

That's right -- agreed. And sure, have now added `FREEZING_POINT_WATER_{IP, SI}`.

Propagated note about using the triple point of water to the other languages and I retested VBA locally. I have re-requested your review. After this we can prepare for the new release. Let me know if you want to consider this a bugfix or not as this to me would be a patch version `2.1.1` -- i.e. point 3 from the list below (https://semver.org/):
```
1. MAJOR version when you make incompatible API changes,
2. MINOR version when you add functionality in a backwards-compatible manner, and
3. PATCH version when you make backwards-compatible bug fixes.
```

**Pull request merged**